x-pack/plugins/endpoint/public/embeddables/resolver/store/middleware.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreStart } from 'kibana/public';
+import { Dispatch, MiddlewareAPI } from 'redux';
+import { ResolverState, ResolverAction } from '../types';
+
+type MiddlewareFactory<S = ResolverState> = (
+  coreStart: CoreStart
+) => (
+  api: MiddlewareAPI<Dispatch<ResolverAction>, S>
+) => (next: Dispatch<ResolverAction>) => (action: ResolverAction) => unknown;
+
+export const resolverMiddlewareFactory: MiddlewareFactory = coreStart => {
+  return api => next => async (action: ResolverAction) => {
+    next(action);
+    if (action.type === 'userChangedSelectedEvent') {
+      // this is brittle as it is going to change assumedly.
+      api.dispatch({ type: 'appRequestedResolverData' });
+      const {
+        coreStart: { http },
+      } = coreStart;
+      //const uniquePid = action.payload.selectedEvent?.value?.source.endgame.data.pid;
+      const uniquePid = '3096';
+      const { lifecycle } = await http.get(`/api/endpoint/resolver/${uniquePid}`, {
```

Comment:
❔ Looks like you could cut a little time out by doing these requests in parallel rather than in a series of `await`s since they don't seem to depend on the previous one. (Or maybe that will change?)